File: 01-basics/scripts/test-hello-world.js
#!/usr/bin/env node

/**
 * n8n Hello World 워크플로우 테스트 스크립트 (Node.js)
 * 사용법: node test-hello-world.js YOUR_WEBHOOK_URL
 */

const https = require('https');
const http = require('http');

// 색상 코드
const colors = {
  reset: '\x1b[0m',
  green: '\x1b[32m',
  blue: '\x1b[34m',
  red: '\x1b[31m'
};

const WEBHOOK_URL = process.argv[2] || 'https://your-n8n-instance.app.n8n.cloud/webhook/hello';

/**
 * HTTP GET 요청 함수
 */
function makeRequest(url) {
  return new Promise((resolve, reject) => {
    const protocol = url.startsWith('https') ? https : http;

    protocol.get(url, (res) => {
      let data = '';

      res.on('data', (chunk) => {
        data += chunk;
      });

      res.on('end', () => {
        try {
          resolve(JSON.parse(data));
        } catch (e) {
          resolve(data);
        }
      });
    }).on('error', (err) => {
      reject(err);
    });
  });
}

/**
 * 테스트 실행
 */
async function runTests() {
  console.log(`${colors.blue}=== n8n Hello World 테스트 ===${colors.reset}\n`);

  // 테스트 1: 기본 요청
  console.log(`${colors.green}테스트 1: 기본 GET 요청${colors.reset}`);
  try {
    const response = await makeRequest(WEBHOOK_URL);
    console.log(JSON.stringify(response, null, 2));
  } catch (error) {
    console.log(`${colors.red}Error: ${error.message}${colors.reset}`);
  }
  console.log('');

  // 테스트 2: Query Parameter 포함
  console.log(`${colors.green}테스트 2: Query Parameter 포함${colors.reset}`);
  try {
    const response = await makeRequest(`${WEBHOOK_URL}?name=홍길동`);
    console.log(JSON.stringify(response, null, 2));
  } catch (error) {
    console.log(`${colors.red}Error: ${error.message}${colors.reset}`);
  }
  console.log('');

  // 테스트 3: 여러 Parameter
  console.log(`${colors.green}테스트 3: 여러 Query Parameters${colors.reset}`);
  try {
    const response = await makeRequest(`${WEBHOOK_URL}?name=홍길동&city=서울`);
    console.log(JSON.stringify(response, null, 2));
  } catch (error) {
    console.log(`${colors.red}Error: ${error.message}${colors.reset}`);
  }
  console.log('');

  console.log(`${colors.blue}=== 테스트 완료 ===${colors.reset}`);
}

// 실행
runTests().catch(console.error);

File: 01-basics/scripts/test-hello-world.py
#!/usr/bin/env python3

"""
n8n Hello World 워크플로우 테스트 스크립트 (Python)
사용법: python test-hello-world.py YOUR_WEBHOOK_URL
"""

import sys
import json
import urllib.request
import urllib.error

# 색상 코드
class Colors:
    GREEN = '\033[0;32m'
    BLUE = '\033[0;34m'
    RED = '\033[0;31m'
    NC = '\033[0m'  # No Color

WEBHOOK_URL = sys.argv[1] if len(sys.argv) > 1 else 'https://your-n8n-instance.app.n8n.cloud/webhook/hello'

def make_request(url):
    """HTTP GET 요청"""
    try:
        with urllib.request.urlopen(url) as response:
            data = response.read()
            return json.loads(data.decode('utf-8'))
    except urllib.error.URLError as e:
        return {'error': str(e)}
    except json.JSONDecodeError:
        return {'error': 'Invalid JSON response'}

def run_tests():
    """테스트 실행"""
    print(f"{Colors.BLUE}=== n8n Hello World 테스트 ==={Colors.NC}\n")

    # 테스트 1: 기본 요청
    print(f"{Colors.GREEN}테스트 1: 기본 GET 요청{Colors.NC}")
    response = make_request(WEBHOOK_URL)
    print(json.dumps(response, indent=2, ensure_ascii=False))
    print()

    # 테스트 2: Query Parameter 포함
    print(f"{Colors.GREEN}테스트 2: Query Parameter 포함{Colors.NC}")
    response = make_request(f"{WEBHOOK_URL}?name=홍길동")
    print(json.dumps(response, indent=2, ensure_ascii=False))
    print()

    # 테스트 3: 여러 Parameter
    print(f"{Colors.GREEN}테스트 3: 여러 Query Parameters{Colors.NC}")
    response = make_request(f"{WEBHOOK_URL}?name=홍길동&city=서울")
    print(json.dumps(response, indent=2, ensure_ascii=False))
    print()

    print(f"{Colors.BLUE}=== 테스트 완료 ==={Colors.NC}")

if __name__ == '__main__':
    run_tests()
